### PR TITLE
Simulate key release

### PR DESCRIPTION
**Pull request by ksqsf**
Support chord typing in rime.

This is by default disabled. To enable, modify `~/.config/fcitx5/conf/macosfrontend.conf`:

```
# False by default
SimulateKeyRelease=True
# in milliseconds, 100ms by default
SimulateKeyReleaseDelay=100
```

### CODE REVIEW

File: macosfrontend/macosfrontend.cpp [hunk 1]
```diff
@@ -205,6 +224,23 @@ bool MacosFrontend::keyEvent(ICUUID uuid, const Key &key, bool isRelease) {
     }
     KeyEvent keyEvent(ic, key, isRelease);
     ic->keyEvent(keyEvent);
+
+    if (simulateKeyRelease_ && !isRelease && !key.isModifier()) {
```
Comment (eagleoflqj) [inline macosfrontend/macosfrontend.cpp:228]: In vscodium I press cmd+n to open a new tab, but with combo pinyin the key release triggers a `er`.

File: macosfrontend/macosfrontend.h [hunk 2]
```diff
@@ -22,12 +25,25 @@ class MacosInputContext;
 enum class PanelShowFlag : int;
 using PanelShowFlags = fcitx::Flags<PanelShowFlag>;
 
+FCITX_CONFIGURATION(MacosFrontendConfig,
+                    fcitx::Option<bool> simulateKeyRelease{
+                        this, "SimulateKeyRelease", _("Simulate key release")};
+                    fcitx::Option<int> simulateKeyReleaseDelay{
+                        this, "SimulateKeyReleaseDelay",
+                        _("Delay of simulated key release in milliseconds"),
+                        100};);
```
Comment (eagleoflqj) [inline macosfrontend/macosfrontend.h:34]: There needs a `IntConstrain` to exclude too small value and too big value.
Do you think 10-1000 make sense?

**Review by eagleoflqj — CHANGES_REQUESTED** [review]
Thanks for enabling chord typing!

File: macosfrontend/macosfrontend.cpp (hunk 1 above)
Comment (ksqsf) [reply macosfrontend/macosfrontend.cpp:228]: Seems like an upstream problem. `chord_processor` returns handled for super+anything. Created a PR: https://github.com/rime/librime/pull/808

**Review by eagleoflqj — APPROVED** [review]

**Pull request merged**